homeassistant/components/overkiz/entity.py

```diff
@@ -27,7 +27,10 @@ def __init__(
         """Initialize the device."""
         super().__init__(coordinator)
         self.device_url = device_url
-        self.base_device_url, *_ = self.device_url.split("#")
+        split_device_url = self.device_url.split("#")
+        self.base_device_url = split_device_url[0]
+        if len(split_device_url) == 2:
+            self.index_device_url = split_device_url[1]
         self.executor = OverkizExecutor(device_url, coordinator)
```

Comment:
What does this code do? Tbh, I don't find this code very clean / readable. And it seems that `self.index_device_url` is not typed / having None as default.

Reply:
In other devices, like somfy_thermostat for instance, we use a constant:

https://github.com/home-assistant/core/blob/551fb449752e1c3f55eb688d24509876020852d1/homeassistant/components/overkiz/climate_entities/somfy_thermostat.py#L51

https://github.com/home-assistant/core/blob/551fb449752e1c3f55eb688d24509876020852d1/homeassistant/components/overkiz/climate_entities/somfy_thermostat.py#L67-L74


I think it might be clearer an less prone to error than doing `n+1` as you do here
```py
        # Temperature sensor use the same base_device_url and use the n+1 index
        self.temperature_device = self.executor.linked_device(
            int(self.index_device_url) + 1
        )
```

Reply:
This type of device don't work like that.
We have to use the climate device index and search for n+1 index, which will be the temperature sensor.
The n+1 is required

Reply:
Does it resolves the conversation ?

Reply:
When [this PR](https://github.com/iMicknl/python-overkiz-api/pull/639) is merged and published, this code will be updated.

Reply:
OK, so, for now we are keeping it like this ?

Reply:
It's working like this, it could be merged as it is.
Up to @iMicknl to decide I guess.